sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusModelFactory.cs

```diff
@@ -504,7 +504,7 @@ public override bool TryAddMessage(ServiceBusMessage message)
             ///
             /// <returns>The set of events as an enumerable of the requested type.</returns>
             ///
-            public override IEnumerable<T> AsEnumerable<T>() => (IEnumerable<T>)_backingStore;
+            public override IReadOnlyCollection<T> AsReadOnly<T>() => new List<T>((IEnumerable<T>)_backingStore);
```

Comment:
This is a necessary evil since any arbitrary `IList<T>` can be passed that might not be directly castable to `IReadOnlyCollection<T>`

Reply:
I agree it's not impactful due to the testing context, but should we add a small bit of logic to streamline when we can?   Something like:

```csharp
public override IReadOnlyCollection<T> AsReadOnly<T>() =>
    _backingStore switch
    { 
        IReadOnlyCollection collection => collection,
        _ => new List<T>((IEnumerable<T>)_backingStore)
    };
```